### DISCUSSION TITLE
after create and rename file keep focus on tree window

### DISCUSSION THREAD

**Discussion by Mr-LLLLL** [nvim-tree/nvim-tree.lua#2750] [Q&A]
is any config setting keep cursor  on tree window when i create or rename file.

**Comment by alex-courtis** [comment]
The cursor moves to the command line in order to prompt for a name; the prompt is obviously required.

You can use a plugin that prompts in a floating window such as https://github.com/rcarriga/nvim-notify

There is no way to keep the focus on the tree window.

**Comment by Mr-LLLLL** [comment]
why? i think quit prompts window or float, will be back to last window.so if i rename a file. will be enter prompt window, and input name.and confirm or quit it. back to last window is obviously. mayby prompt window quit will jump to other window is random?

**Reply by alex-courtis** [reply]
I'm sorry, it is clear that I don't understand the question or your use case.

Perhaps you might record a video with notes as to what happens and what you expect e.g.

0:15 renamed file
0:17 xxxx focused, yyyy should be focused
0.20 typed new name
....

**Comment by Mr-LLLLL** [comment]
i am sorry for this. i found the bug is not nvim-tree caused. thank your help.

**Reply by alex-courtis** [reply]
That's great! What was the bug and how did you resolve it?

**Reply by Mr-LLLLL** [reply]
some plug listening InsertLeave event, and callback vim.cmd(). so i create or rename file will i trigger this callback. than occurs.but i dont know why happen this.

**Reply by Mr-LLLLL** [reply]
maybe when InsertLeave trigger callback, cursor will be back to rename Input window, but the rename Input window is quit. so cursor jumped window of hold max window number?
